### PR TITLE
Page builder blocks v2 UI list page blocks

### PR DESCRIPTION
**Pull request by neatbyte-vnobis**
## Changes
Add Page Blocks page to Page Builder UI app

## How Has This Been Tested?
Manual testing

## Documentation
None

### CODE REVIEW

File: packages/app-page-builder/src/admin/plugins/routes.tsx [hunk 1]
```diff
@@ -111,6 +112,24 @@ const plugins: RoutePlugin[] = [
...
+        name: "route-pb-page-blocks",
+        type: "route",
+        route: (
+            <Route
+                exact
+                path="/page-builder/page-blocks"
+                render={() => (
+                    <SecureRoute permission={ROLE_PB_BLOCK}>
+                        <AdminLayout>
+                            <Helmet title={"Page Builder - Page Blocks"} />
```
Comment (adrians5j) [inline packages/app-page-builder/src/admin/plugins/routes.tsx]: On the frontend / where we're just printing labels/titles, we can use just `Blocks` label (so, `Page Builder - Blocks`.

File: packages/app-page-builder/src/PageBuilder.tsx [hunk 2]
```diff
@@ -57,13 +57,18 @@ const PageBuilderMenu: React.FC = () => {
                         />
                     </HasPermission>
                 </Menu>
-                <Menu name="pageBuilder.blocks" label={"Blocks"}>
+                <Menu name="pageBuilder.blocks" label={"Page Blocks"}>
```
Comment (adrians5j) [inline packages/app-page-builder/src/PageBuilder.tsx]: Let's do just `Blocks` here as well.

File: packages/app-page-builder/src/admin/views/PageBlocks/BlocksByCategoriesDataList.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
+import React, { useCallback, useMemo, useState } from "react";
+import { i18n } from "@webiny/app/i18n";
+import { useRouter } from "@webiny/react-router";
+import { useQuery } from "@apollo/react-hooks";
```
Comment (adrians5j) [inline packages/app-page-builder/src/admin/views/PageBlocks/BlocksByCategoriesDataList.tsx:4]: Can you clarify please, what is this view? Can you provide a screenshot?

**Review by adrians5j — CHANGES_REQUESTED** [review]
Just a couple of comments.

File: packages/app-page-builder/src/admin/views/PageBlocks/BlocksByCategoriesDataList.tsx (hunk 3 above)
Comment (neatbyte-vnobis) [reply packages/app-page-builder/src/admin/views/PageBlocks/BlocksByCategoriesDataList.tsx:4]: image (12)
Here is a screenshot

Design link - https://miro.com/app/board/uXjVO1XGqwg=/?share_link_id=951919244316

**Comment by adrians5j** [comment]
Was about to merget this, but I noticed one thing.

@neatbyte-vnobis , should this be "Search categories" here? Maybe "Search blocks" ?

image

**Comment by neatbyte-vnobis** [comment]
@adrians5j 
There we are searching not blocks but categories (left panel list)
But here I may be wrong...
@SvenAlHamad can you clarify?

Design (https://miro.com/app/board/uXjVO1XGqwg=/):
image

**Comment by SvenAlHamad** [comment]
We are searching for blocks, that’s correct.

**Comment by neatbyte-vnobis** [comment]
@SvenAlHamad @adrians5j 
In such case left panel should be affected too, am I right?
So if we search for "Heading" and there are 3 categories with such blocks - we show 3 categories and only blocks containing "Heading" word.
In such case:
1) other categories are not shown
2) counter bellow categories will show number of blocks that are matching search query

I will have to rewrite quite a few things to do it:
now left and right parts of the page are more or less independent (as other similar pages aka categories)
but with such change I need to move all GraphQL queries to parent component (so left and right parts will be using a data from it)
@adrians5j maybe I am missing an easy solution here? Feel free to give me pointer.

**Comment by adrians5j** [comment]
@neatbyte-vnobis In all of our views, these search inputs are doing a search over the entity the user's currently looking at.

For example, on the pages view, pages are being searched. On the page categories view, page categories are being searched.

So, if we were to follow the above examples and wanted to stay consistent, typing into the search field would do a search against existing page blocks.

But!

Now when I look at it, I can see that this view **IS** a bit different than the other ones we have. Instead of just showing blocks in the list, we're showing block categories. Once clicked, the blocks are shown on the right.

And now I do think we need to think about this.

I do agree that this should search for blocks because the user will most probably want to do that (instead of searching for block categories). But I can also see that, in the provided designs, the label said `SEARCH CATEGORIES`, so I can understand that this might require some changes on your side @neatbyte-vnobis.

So.... what I'd do here now is....... 

@SvenAlHamad - please confirm the direction into which we want to move. Do we want to leave it as-is (and save potentially some time), or do we want to revisit this now, and make the search bar do a search over blocks?

Hopefully, I didn't miss anything (like some docs), but yeah, let's first sync on what exactly we want to have here.

**Comment by SvenAlHamad** [comment]
Ok, let's leave it so it's searching blocks if that's what's currently implemented. Once we start using it a bit I think we'll have then one milestone addressing the initial feedback and some additional polishes.

**Comment by adrians5j** [comment]
@SvenAlHamad 

Actually, the way it's currently implemented is that it's searching for block categories, not blocks.

Miro:
image

Real app:
image

I'd still leave it as is, so we can address it then in the MS you mentioned.

**Comment by SvenAlHamad** [comment]
Sorry, I miss typed in my previous comment. I DO understand it's searching "block categories" and I'm happy to leave it like that for now.

**Comment by adrians5j** [comment]
@neatbyte-vnobis feel free to merge this one. 

Thank you!

**Review by adrians5j — APPROVED** [review]

**Pull request merged**